server/src/main/java/org/elasticsearch/index/codec/FilterCodecService.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.codec;
+
+import org.apache.lucene.codecs.Codec;
+
+public class FilterCodecService implements CodecSupplier {
```

Comment:
This is not strictly needed here, it will be used by extending code, but thought it would be nice to have (mimicking the various `Filter*` implementations in Lucene). But I'm OK to remove it if you think it does not belong here.

Reply:
Given this could be just a lambda, I'd suggest to remove it from ES. But no strong opinion on this.